actors/evm/src/interpreter/instructions/call.rs

```diff
@@ -164,79 +163,72 @@ pub fn call<BS: Blockstore, RT: Runtime<BS>>(
             let dst_addr: EthAddress = dst.try_into()?;
             let dst_addr: Address = dst_addr.try_into()?;
 
-            // Special casing for embryo/non-existent actors: we just do a SEND (method 0)
+            // Special casing for account/embryo/non-existent actors: we just do a SEND (method 0)
             // which allows us to transfer funds (and create embryos)
-            let is_embryonic = if dst_addr.protocol() == AddressProtocol::ID {
-                // sanity check: this shouldn't be an ID address, as you can't predict
-                // what actor is gonna sit there.
-                false
-            } else if let Some(actor_id) = system.rt.resolve_address(&dst_addr) {
-                if let Some(cid) = system.rt.get_actor_code_cid(&actor_id) {
-                    system.rt.resolve_builtin_actor_type(&cid) == Some(Type::Embryo)
-                } else {
-                    true
-                }
-            } else {
-                true
-            };
-
-            let call_result = if is_embryonic {
-                system.send(
-                    &dst_addr,
-                    0,
-                    // we still send the input, even thought it will be ignored, for debugging
-                    // purposes
-                    RawBytes::serialize(BytesSer(input_data))?,
-                    TokenAmount::from(&value),
-                )
-            } else {
-                match kind {
-                    CallKind::Call => system.send(
+            let target_actor_code = system
+                .rt
+                .resolve_address(&dst_addr)
+                .and_then(|actor_id| system.rt.get_actor_code_cid(&actor_id))
+                .and_then(|cid| system.rt.resolve_builtin_actor_type(&cid));
+
+            let call_result =
+                if matches!(target_actor_code, Some(Type::Embryo | Type::Account) | None) {
+                    system.send(
```

Comment:
Value transfers are disallowed if this actor was executing in read only (i.e. if this actor was called with STATICCALL).

Reply:
Yes, but that's rejected above? I guess we should have another "if it doesn't exist, don't send" check to prevent automatic actor deployment when in read-only mode (which is why we _really_ need system support).